nwb_conversion_tools/utils/si090nwbephyswriter.py

```diff
@@ -0,0 +1,191 @@
+
+    def _get_channel_property_values(self, prop):
+        if prop == "location":
+            try:
+                return self.recording.get_channel_locations()
+            except:
+                return np.nan * np.ones([len(self._get_channel_ids()), 2])
+        elif prop == "gain":
+            return self.recording.get_channel_gains()
+        elif prop == "offset":
+            return self.recording.get_channel_offsets()
+        elif prop == "group":
+            if self.recording.get_property("group_name") is not None:
+                return self.recording.get_property("group_name")
+            if self.recording.get_channel_groups() is None:
+                return np.zeros(len(self._get_channel_ids()))
+            else:
+                return self.recording.get_channel_groups()
+        elif prop == "name":
+            ids = self.recording.get_channel_ids()
+            if isinstance(ids[0], str):
+                return ids
+        else:
+            prop_values = self.recording.get_property(prop)
+            return self._check_valid_property(prop_values)
+
+    def _get_recording_times(self, segment_index=0):
+        return np.arange(
+            0,
+            self._get_num_frames(segment_index=segment_index) * self._get_sampling_frequency(),
+            self._get_sampling_frequency(),
+        )
+
+    def _get_unit_feature_names(self):
+        return []
+
+    def _get_unit_feature_values(self, prop):
+        return []
+
+    def _get_unit_spike_train_ids(self, unit_id, start_frame=None, end_frame=None, segment_index=None):
+        return self.sorting.get_unit_spike_train(
+            unit_id, start_frame=start_frame, end_frame=end_frame, segment_index=segment_index
+        )
+
+    def _get_unit_spike_train_times(self, unit_id, segment_index=0):
+        return self._get_unit_spike_train_ids(unit_id, segment_index) / self._get_unit_sampling_frequency()
+
+    def _get_unit_property_names(self):
+        properties = self.sorting.get_property_keys()
+        if self.waveforms is not None:
+            if "max_channel" not in properties:
+                properties.append("max_channel")
+        return properties
+
+    def _get_unit_property_values(self, prop):
+        prop_values = self.sorting.get_property(prop)
+        if self.waveforms is not None:
+            if prop_values is None and prop == "max_channel":
+                from spikeinterface.toolkit import get_template_extremum_channel
+
+                channels_dict = get_template_extremum_channel(self.waveforms)
+                prop_values = [channels_dict.get(id, np.nan) for id in self._get_unit_ids()]
+        return self._check_valid_property(prop_values)
+
+    def _get_unit_waveforms_templates(self, unit_id, mode="average"):
+        return self.waveforms.get_template(unit_id, mode=mode)
+
+    def add_epochs(self):
+        return
+
+
+def create_si090_example():
```

Comment:
should this be here?

Reply:
moved the methods https://github.com/catalystneuro/nwb-conversion-tools/pull/230/commits/083dafa2a3eae16ac099905a1508e28f3c97bf52